From 1b06d3b9ab636321f4563d07584e88193bdbb245 Mon Sep 17 00:00:00 2001
From: Lars-Peter Clausen <lars@metafoo.de>
Date: Wed, 23 Nov 2011 11:40:40 +0100
Subject: ASoC: Constify snd_soc_dai_ops structs

Commit 1ee46ebd("ASoC: Make the DAI ops constant in the DAI structure")
introduced the possibility to have constant DAI ops structures, yet this is
barley used in both existing drivers and also new drivers being submitted,
although none of them modifies its DAI ops structure. The later is not
surprising since existing drivers are often used as templates for new drivers.
So this patch just constifies all existing snd_soc_dai_ops structs to eliminate
the issue altogether.

The patch was generated with the following coccinelle semantic patch:
// <smpl>
@@
identifier ops;
@@
-struct snd_soc_dai_ops ops =
+const struct snd_soc_dai_ops ops =
{ ... };
// </smpl>

Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>
Signed-off-by: Mark Brown <broonie@opensource.wolfsonmicro.com>
(cherry picked from commit 85e7652d89293a6dab42bfd31f276f8bc072d4c5)

Commits:

	sound/soc/atmel/atmel_ssc_dai.c
	sound/soc/au1x/psc-ac97.c
	sound/soc/au1x/psc-i2s.c
	sound/soc/blackfin/bf5xx-i2s.c
	sound/soc/blackfin/bf5xx-tdm.c
	sound/soc/codecs/88pm860x-codec.c
	sound/soc/codecs/ac97.c
	sound/soc/codecs/ad1836.c
	sound/soc/codecs/ad193x.c
	sound/soc/codecs/adau1373.c
	sound/soc/codecs/adau1701.c
	sound/soc/codecs/adav80x.c
	sound/soc/codecs/ak4104.c
	sound/soc/codecs/ak4535.c
	sound/soc/codecs/ak4641.c
	sound/soc/codecs/ak4642.c
	sound/soc/codecs/ak4671.c
	sound/soc/codecs/alc5623.c
	sound/soc/codecs/alc5632.c
	sound/soc/codecs/cq93vc.c
	sound/soc/codecs/cs4270.c
	sound/soc/codecs/cs4271.c
	sound/soc/codecs/cs42l51.c
	sound/soc/codecs/cs42l73.c
	sound/soc/codecs/da7210.c
	sound/soc/codecs/jz4740.c
	sound/soc/codecs/max98088.c
	sound/soc/codecs/max98095.c
	sound/soc/codecs/max9850.c
	sound/soc/codecs/rt5631.c
	sound/soc/codecs/sgtl5000.c
	sound/soc/codecs/sn95031.c
	sound/soc/codecs/ssm2602.c
	sound/soc/codecs/sta32x.c
	sound/soc/codecs/stac9766.c
	sound/soc/codecs/tlv320aic23.c
	sound/soc/codecs/tlv320aic26.c
	sound/soc/codecs/tlv320aic32x4.c
	sound/soc/codecs/tlv320aic3x.c
	sound/soc/codecs/tlv320dac33.c
	sound/soc/codecs/twl4030.c
	sound/soc/codecs/twl6040.c
	sound/soc/codecs/uda134x.c
	sound/soc/codecs/uda1380.c
	sound/soc/codecs/wl1273.c
	sound/soc/codecs/wm5100.c
	sound/soc/codecs/wm8350.c
	sound/soc/codecs/wm8400.c
	sound/soc/codecs/wm8510.c
	sound/soc/codecs/wm8523.c
	sound/soc/codecs/wm8580.c
	sound/soc/codecs/wm8711.c
	sound/soc/codecs/wm8728.c
	sound/soc/codecs/wm8731.c
	sound/soc/codecs/wm8737.c
	sound/soc/codecs/wm8741.c
	sound/soc/codecs/wm8750.c
	sound/soc/codecs/wm8753.c
	sound/soc/codecs/wm8770.c
	sound/soc/codecs/wm8776.c
	sound/soc/codecs/wm8804.c
	sound/soc/codecs/wm8900.c
	sound/soc/codecs/wm8903.c
	sound/soc/codecs/wm8904.c
	sound/soc/codecs/wm8940.c
	sound/soc/codecs/wm8955.c
	sound/soc/codecs/wm8960.c
	sound/soc/codecs/wm8961.c
	sound/soc/codecs/wm8962.c
	sound/soc/codecs/wm8971.c
	sound/soc/codecs/wm8974.c
	sound/soc/codecs/wm8978.c
	sound/soc/codecs/wm8983.c
	sound/soc/codecs/wm8985.c
	sound/soc/codecs/wm8988.c
	sound/soc/codecs/wm8990.c
	sound/soc/codecs/wm8991.c
	sound/soc/codecs/wm8993.c
	sound/soc/codecs/wm8994.c
	sound/soc/codecs/wm8995.c
	sound/soc/codecs/wm8996.c
	sound/soc/codecs/wm9081.c
	sound/soc/codecs/wm9705.c
	sound/soc/codecs/wm9712.c
	sound/soc/codecs/wm9713.c
	sound/soc/davinci/davinci-i2s.c
	sound/soc/davinci/davinci-mcasp.c
	sound/soc/davinci/davinci-vcif.c
	sound/soc/ep93xx/ep93xx-ac97.c
	sound/soc/ep93xx/ep93xx-i2s.c
	sound/soc/fsl/fsl_ssi.c
	sound/soc/fsl/mpc5200_psc_ac97.c
	sound/soc/fsl/mpc5200_psc_i2s.c
	sound/soc/imx/imx-ssi.c
	sound/soc/jz4740/jz4740-i2s.c
	sound/soc/kirkwood/kirkwood-i2s.c
	sound/soc/mxs/mxs-saif.c
	sound/soc/nuc900/nuc900-ac97.c
	sound/soc/omap/ams-delta.c
	sound/soc/omap/omap-hdmi.c
	sound/soc/omap/omap-mcbsp.c
	sound/soc/omap/omap-mcpdm.c
	sound/soc/pxa/pxa-ssp.c
	sound/soc/pxa/pxa2xx-ac97.c
	sound/soc/pxa/pxa2xx-i2s.c
	sound/soc/s6000/s6000-i2s.c
	sound/soc/samsung/ac97.c
	sound/soc/samsung/i2s.c
	sound/soc/samsung/pcm.c
	sound/soc/samsung/s3c2412-i2s.c
	sound/soc/samsung/s3c24xx-i2s.c
	sound/soc/samsung/spdif.c
	sound/soc/sh/hac.c
	sound/soc/sh/siu_dai.c
	sound/soc/sh/ssi.c
	sound/soc/soc-core.c
	sound/soc/tegra/tegra_i2s.c
	sound/soc/tegra/tegra_spdif.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 sound/soc/sh/fsi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/sh/fsi.c b/sound/soc/sh/fsi.c
index 99ed610..aa30330 100644
--- a/sound/soc/sh/fsi.c
+++ b/sound/soc/sh/fsi.c
@@ -1096,7 +1096,7 @@ static int fsi_dai_hw_params(struct snd_pcm_substream *substream,
 	return ret;
 }
 
-static struct snd_soc_dai_ops fsi_dai_ops = {
+static const struct snd_soc_dai_ops fsi_dai_ops = {
 	.startup	= fsi_dai_startup,
 	.shutdown	= fsi_dai_shutdown,
 	.trigger	= fsi_dai_trigger,
-- 
1.7.10.2.565.gbd578b5

